Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/factory.go

```diff
@@ -209,6 +209,9 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				}
 				return &clientSwaggerSchema{client, api.Scheme}, nil
 			}
+			if filenameFlag := flags.Lookup("filename"); filenameFlag != nil && filenameFlag.Changed {
```

Comment:
> This forces filename validation on all objects processed

Yeah, that's the caveat of doing it this way. Today `create`, `update`, `rollingupdate`, etc all use the resource builder just once. But this could introduce an issue in future.

Any suggestion about another way to use the validator, or should we just switch the approach, e.g. in `ToJSON`?